spec/forms/flood_risk_engine/steps/local_authority_address_form_spec.rb

```diff
@@ -58,35 +40,33 @@ def form_requires_address_lookup
         }
 
         it "is valid when valid UK UPRN supplied via drop down but no street_address present" do
-          form_requires_address_no_street_lookup do
-            expect(form.validate(valid_but_no_street_address)).to eq true
-          end
+          mock_ea_address_lookup_find_by_uprn
+          expect(form.validate(valid_but_no_street_address)).to eq true
         end
 
         describe "Save" do
           it "saves a valid address on enrollment" do
-            form_requires_address_lookup do
-              form.validate(valid_attributes)
-              expect(form.save).to eq true
-            end
+            mock_ea_address_lookup_find_by_uprn
+            form.validate(valid_attributes)
+            expect(form.save).to eq true
 
-            expect(subject.model.postcode).to eq(valid_attributes[:post_code])
+            expect(form.model.postcode).to eq(valid_attributes[:post_code])
 
-            address = subject.enrollment.reload.organisation.primary_address
+            address = form.enrollment.reload.organisation.primary_address
 
             expected_attributes = {
               address_type: "primary",
               premises: "HORIZON HOUSE",
               street_address: "DEANERY ROAD",
-              locality: nil,
+              locality: "SOMEWHERE",
```

Comment:
seems better for test data to behave exactly as returned by address service - looks like the mock data set has been manually altered